clients/src/main/java/org/apache/kafka/clients/admin/internals/DeleteConsumerGroupOffsetsHandler.java

```diff
@@ -97,54 +108,79 @@ public String apiName() {
         Set<CoordinatorKey> groupIds,
         AbstractResponse abstractResponse
     ) {
+        validateKeys(groupIds);
+
         final OffsetDeleteResponse response = (OffsetDeleteResponse) abstractResponse;
-        Map<CoordinatorKey, Map<TopicPartition, Errors>> completed = new HashMap<>();
-        Map<CoordinatorKey, Throwable> failed = new HashMap<>();
-        List<CoordinatorKey> unmapped = new ArrayList<>();
+        final Map<CoordinatorKey, Map<TopicPartition, Errors>> completed = new HashMap<>();
+        final Map<CoordinatorKey, Throwable> failed = new HashMap<>();
+        final Set<CoordinatorKey> groupsToUnmap = new HashSet<>();
+        final Set<CoordinatorKey> groupsToRetry = new HashSet<>();
 
         final Errors error = Errors.forCode(response.data().errorCode());
         if (error != Errors.NONE) {
-            handleError(groupId, error, failed, unmapped);
+            handleGroupError(groupId, error, failed, groupsToUnmap, groupsToRetry);
```

Comment:
It seems that `groupsToRetry` is not really necessary in this case. Moreover, we could directly return in the branch as we don't expect errors in the partitions.

```
if (error != Errors.NONE) {
  final Map<CoordinatorKey, Throwable> failed = new HashMap<>();
  final Set<CoordinatorKey> groupsToUnmap = new HashSet<>();

  handleGroupError(groupId, error, failed, groupsToUnmap);

  return new ApiResult<>(Collections.emptyMap(), failed, new ArrayList<>(groupsToUnmap);
}
```

`groupId` will be either in `failed` or in `groupsToUnmap` after the call to `handleGroupError`.

Reply:
good suggestion! Updated!